pkg/controlsvc/reload.go

```diff
@@ -88,53 +71,63 @@ func parseConfigForReload(filename string, checkReload bool) error {
 			return err
 		}
 		cfg := string(cfgBytes)
-		if !isReloadable(cfg) {
-			if checkReload {
-				if _, ok := cfgNotReloadable[cfg]; !ok {
-					action := getActionKeyword(cfg)
-
-					return fmt.Errorf("a non-reloadable config action '%s' was modified or added. Must restart receptor for these changes to take effect", action)
-				}
-				cfgNotReloadable[cfg] = true
-			} else {
-				cfgNotReloadable[cfg] = false
-			}
-		}
+		cfgMap[cfg] = struct{}{}
 	}
 
 	return nil
 }
 
-func cfgAbsent() error {
-	// checks to see if any item in cfgNotReloadable has a value of false,
-	// if so, that means an unreloadable item has been removed from the config
-	defer func() {
-		for k := range cfgNotReloadable {
-			cfgNotReloadable[k] = false
+func checkReload() error {
+	// Determine which items from the old config have been added or modified
+	for cfg := range cfgNext {
+		action := getActionKeyword(cfg)
+		_, isReloadable := reloadableActions[action]
```

Comment:
This checks if `action` is a key in `reloadableActions`, not if it is reloadable. Other name perhaps?